src/Dictionaries/HashedArrayDictionary.cpp

```diff
@@ -969,6 +971,9 @@ void registerDictionaryArrayHashed(DictionaryFactory & factory)
 
         HashedArrayDictionaryStorageConfiguration configuration{require_nonempty, dict_lifetime, static_cast<size_t>(shards)};
 
+        if (source_ptr->hasUpdateField() && shards > 1)
```

Comment:
We have the same for `hashed` dictionary, but it was missed for `hashed_array`. If we specify both `update_field` and `SHARDS` dictionary will not work properly and will return wrong results.